src/main/java/org/alfresco/utility/data/DataContent.java

```diff
@@ -276,10 +472,161 @@ public FileModel createContent(FileModel fileModel) throws DataPreparationExcept
     }
 
     /**
-     * @param fullPath - the full path to CMIS object
-     * @param userModel
-     * @throws TestConfigurationException
+     * This is the entry point of the createContent() method to make REST API or CMIS call
      */
+    public FileModel createContent(DocumentType documentType) throws DataPreparationException
+    {
+        AlfrescoHttpClient client = alfrescoHttpClientFactory.getObject();
+
+        if(client.getAlfVersion() >= 5.2)
+        {
+            FileModel fileModel = new FileModel(RandomData.getRandomName("file"));
+            FileModel createFile = createContentDocTypeV1Api(client, fileModel, documentType);

```

Comment:
again, error handling in these cases, if something goes wrong.. have we considered all the paths and outcomes of the REST calls ? 
how is a failure handled by the next method call ? 
how is the failure reported to the calling methods of this method ?

Reply:
Changed, let's talk to see if the changes are appropriate